mittab/apps/tab/tab_card.py

```diff
@@ -0,0 +1,284 @@
+        4: ("LF", "WF"),
+        5: ("AD", "AD"),
+        6: ("AW", "AW"),
+    }
+    return victor_map[victor_code][side]
+
+def get_dstats(round_obj, deb1, deb2, iron_man):
+    """Taken from original tab card function.
+    Pretty sure there is a good refactor here but seems hard to test and worried about breaking something"""
+    dstat1 = [
+            k for k in RoundStats.objects.filter(debater=deb1).filter(
+                round=round_obj).all()
+        ]
+    dstat2 = []
+    if not iron_man:
+        dstat2 = [
+            k for k in RoundStats.objects.filter(debater=deb2).filter(
+                round=round_obj).all()
+        ]
+    blank_rs = RoundStats(debater=deb1, round=round_obj, speaks=0, ranks=0)
+    while len(dstat1) + len(dstat2) < 2:
+        # Something is wrong with our data, but we don't want to crash
+        dstat1.append(blank_rs)
+    if not dstat2 and not dstat1:
+        return None, None
+    if not dstat2:
+        dstat1, dstat2 = dstat1[0], dstat1[1]
+    elif not dstat1:
+        dstat1, dstat2 = dstat2[0], dstat2[1]
+    else:
+        dstat1, dstat2 = dstat1[0], dstat2[0]
+    return dstat1, dstat2
+
+def json_get_round(round_obj, team, deb1, deb2):
+    jsonRound = {
+        "round_number": round_obj.round_number,
+        "round_id": round_obj.pk,
+        "side": "G" if round_obj.gov_team == team else "O",
+        "result": get_victor_label(round_obj.victor, "G" if round_obj.gov_team == team else "O"),
+        "chair": round_obj.chair.name,
+        "wings": [judge.name for judge in round_obj.judges.exclude(pk=round_obj.chair.pk)]
+    }
+    
+    opponent = round_obj.opp_team if round_obj.gov_team == team else round_obj.gov_team
+    if opponent:
+        opponent_debaters = list(opponent.debaters.all())
+        jsonRound["opponent"] = {
+            "name": opponent.get_team_code(),
+            "school": opponent.school.name,
+            "debater1": opponent_debaters[0].name if opponent_debaters else None,
+            "debater2": opponent_debaters[1].name if len(opponent_debaters) > 1 else None,
+        }
+
+    jsonRound["debater1"] = list(
+        (stat.speaks, stat.ranks) 
+        for stat in RoundStats.objects.filter(debater=deb1, round=round_obj)
+    )
+    jsonRound["debater2"] = list(
+        (stat.speaks, stat.ranks) 
+        for stat in RoundStats.objects.filter(debater=deb2, round=round_obj)
+    ) if deb2 else []
+
+    try:
+        bye_round = Bye.objects.get(bye_team=team).round_number
+        jsonRound[bye_round - 1][bye_round]
+    except Bye.DoesNotExist:
+        pass
+
+    return jsonRound
+
+class JSONDecimalEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, Decimal):
+            return float(obj)
+        return json.JSONEncoder.default(self, obj)
+    
+def get_all_json_data():
+    all_tab_cards_data = {}
+    
+    # Prefetch related data to reduce database queries
+    teams = Team.objects.prefetch_related(
```

Comment:
I think we wanna fetch judges here as well, lemme test locally to see how to do that